MoreLinq/ScanBy.cs

```diff
@@ -0,0 +1,130 @@
+        /// source sequence, and the current state for that key.
+        /// </summary>
+        /// <typeparam name="TSource">Type of the elements of the source sequence.</typeparam>
+        /// <typeparam name="TKey">Type of the projected key.</typeparam>
+        /// <typeparam name="TState">Type of state elements.</typeparam>
+        /// <param name="source">Source sequence.</param>
+        /// <param name="keySelector">Function that transforms each item of source sequence into a key.</param>
+        /// <param name="seedSelector">Function that defines the initial state for that key.</param>
+        /// <param name="accumulator">Function that defines the current state for that key.</param>
+        /// <returns>
+        /// Returns a sequence of states per key, that is, the key of each element of the
+        /// source sequence, and the current state for that key.
+        /// </returns>
+
+        public static IEnumerable<KeyValuePair<TKey, TState>> ScanBy<TSource, TKey, TState>(
+            this IEnumerable<TSource> source,
+            Func<TSource, TKey> keySelector,
+            Func<TKey, TState> seedSelector,
+            Func<TSource, TKey, TState, TState> accumulator)
+        {
+            return source.ScanBy(keySelector, seedSelector, accumulator, null);
+        }
+
+        /// <summary>
+        /// Scans the source sequence and performs accumulations of state for each group of keys.
+        /// Returns a sequence of states per key, that is, the key of each element of the
+        /// source sequence, and the current state for that key.
+        /// </summary>
+        /// <typeparam name="TSource">Type of the elements of the source sequence.</typeparam>
+        /// <typeparam name="TKey">Type of the projected key.</typeparam>
+        /// <typeparam name="TState">Type of state elements.</typeparam>
+        /// <param name="source">Source sequence.</param>
+        /// <param name="keySelector">Function that transforms each item of source sequence into a key.</param>
+        /// <param name="seedSelector">Function that defines the initial state for that key.</param>
+        /// <param name="accumulator">Function that defines the current state for that key.</param>
+        /// <param name="comparer">The equality comparer to use to determine whether or not keys are equal.
+        /// If null, the default equality comparer for <typeparamref name="TSource"/> is used.</param>
+        /// <returns>
+        /// Returns a sequence of states per key, that is, the key of each element of the
+        /// source sequence, and the current state for that key.
+        /// </returns>
+
+        public static IEnumerable<KeyValuePair<TKey, TState>> ScanBy<TSource, TKey, TState>(
+            this IEnumerable<TSource> source,
+            Func<TSource, TKey> keySelector,
+            Func<TKey, TState> seedSelector,
+            Func<TSource, TKey, TState, TState> accumulator,
```

Comment:
For consistency with `Scan`, the type arguments of `accumulator` should be reversed to be `Func<TState, TKey, TSource, TState>`.

Reply:
adjusted in [#d6f5755](https://github.com/morelinq/MoreLINQ/pull/572/commits/d6f5755ad1aea7b3d4fb9d702c58e55a171c094d)